File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/CBuilder/ColumnsShareDemo/ColumnsShareDemoData.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "ColumnsShareDemoData.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxClasses"
#pragma link "cxGridTableView"
#pragma link "cxStyles"
#pragma link "cxGridCardView"
#pragma resource "*.dfm"
TColumnsShareDemoMainDM *ColumnsShareDemoMainDM;
//---------------------------------------------------------------------------
__fastcall TColumnsShareDemoMainDM::TColumnsShareDemoMainDM(TComponent* Owner)
  : TDataModule(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TColumnsShareDemoMainDM::tblUsersCalcFields(TDataSet *DataSet)
{
  tblUsersUserName->AsString = tblUsersFNAME->AsString +  " " + tblUsersLNAME->AsString;
}
//---------------------------------------------------------------------------
void __fastcall TColumnsShareDemoMainDM::DataModuleCreate(TObject *Sender)
{
	String APath = ExtractFilePath(Application->ExeName) + "..\\..\\Data\\";
	tblDepartments->LoadFromFile(APath + "Departments.xml");
	tblItems->LoadFromFile(APath + "Items.xml");
	tblProjects->LoadFromFile(APath + "Projects.xml");
	tblProjectTeam->LoadFromFile(APath + "ProjectTeam.xml");
	tblUsers->LoadFromFile(APath + "Users.xml");
}

File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/Delphi/ColumnsShareDemo/ColumnsShareDemoData.pas
unit ColumnsShareDemoData;

interface

uses
  SysUtils, Classes, DB, cxStyles, cxClasses, cxGridTableView, Forms,
  cxGridCardView, ImgList, Controls, DBClient, MidasLib;

type
  TColumnsShareDemoMainDM = class(TDataModule)
    tblProjects: TClientDataSet;
    dsProjects: TDataSource;
    dsItems: TDataSource;
    tblItems: TClientDataSet;
    tblUsers: TClientDataSet;
    dsUsers: TDataSource;
    dsProjectTeam: TDataSource;
    tblProjectTeam: TClientDataSet;
    dsDepartments: TDataSource;
    tblDepartments: TClientDataSet;
    tblUsersID: TAutoIncField;
    tblUsersFNAME: TStringField;
    tblUsersMNAME: TStringField;
    tblUsersLNAME: TStringField;
    tblUsersCOUNTRY: TStringField;
    tblUsersPOSTALCODE: TStringField;
    tblUsersCITY: TStringField;
    tblUsersADDRESS: TStringField;
    tblUsersPHONE: TStringField;
    tblUsersFAX: TStringField;
    tblUsersEMAIL: TStringField;
    tblUsersHOMEPAGE: TStringField;
    tblUsersDEPARTMENTID: TIntegerField;
    tblUsersUserName: TStringField;
    tblProjectTeamID: TAutoIncField;
    tblProjectTeamPROJECTID: TIntegerField;
    tblProjectTeamUSERID: TIntegerField;
    tblProjectTeamFUNCTION: TStringField;
    tblProjectsID: TAutoIncField;
    tblProjectsNAME: TStringField;
    tblProjectsMANAGERID: TIntegerField;
    tblItemsID: TAutoIncField;
    tblItemsNAME: TStringField;
    tblItemsTYPE: TBooleanField;
    tblItemsPROJECTID: TIntegerField;
    tblItemsPRIORITY: TSmallintField;
    tblItemsSTATUS: TSmallintField;
    tblItemsCREATORID: TIntegerField;
    tblItemsCREATEDDATE: TDateTimeField;
    tblItemsOWNERID: TIntegerField;
    tblItemsLASTMODIFIEDDATE: TDateTimeField;
    tblItemsFIXEDDATE: TDateTimeField;
    tblItemsDESCRIPTION: TMemoField;
    tblItemsRESOLUTION: TMemoField;
    imStat: TImageList;
    tblUsersDepartment: TStringField;
    procedure tblUsersCalcFields(DataSet: TDataSet);
    procedure DataModuleCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  ColumnsShareDemoMainDM: TColumnsShareDemoMainDM;

implementation

{$R *.dfm}

procedure TColumnsShareDemoMainDM.tblUsersCalcFields(DataSet: TDataSet);
begin
  tblUsersUserName.AsString := tblUsersFNAME.AsString + ' ' + tblUsersLNAME.AsString;
end;

procedure TColumnsShareDemoMainDM.DataModuleCreate(Sender: TObject);
var
  APath: string;
begin
  APath := ExtractFilePath(Application.ExeName) + '..\..\Data\';
  tblDepartments.LoadFromFile(APath + 'Departments.xml');
  tblItems.LoadFromFile(APath + 'Items.xml');
  tblProjects.LoadFromFile(APath + 'Projects.xml');
  tblProjectTeam.LoadFromFile(APath + 'ProjectTeam.xml');
  tblUsers.LoadFromFile(APath + 'Users.xml');
end;

end.
